### ISSUE TITLE
Eval bug: Dflash decode error

### ISSUE THREAD

**Issue by MasterLivens** [spiritbuun/buun-llama-cpp#19]
### Name and Version

llama-cli --version
ggml_cuda_init: found 1 CUDA devices (Total VRAM: 24073 MiB):
  Device 0: NVIDIA GeForce RTX 4090, compute capability 8.9, VMM: yes, VRAM: 24073 MiB
version: 8891 (22464d084)
built with GNU 13.3.0 for Linux x86_64

### Operating systems

Linux

### GGML backends

CPU, CUDA

### Hardware

1X RTX4090 24GB

### Models

unsloth Qwen3.6 27B Q4_K_M

### Problem description & steps to reproduce

I tried to run with this cmd, it loaded model succefully, but I got error on web chat, even just say "hello", It doesn't response correctlly, it shows me that it is reasioning, but nothing output.

llama-server -m /home/livens/.cache/huggingface/hub/models--unsloth--Qwen3.6-27B-GGUF/snapshots/82d411acf4a06cfb8d9b073a5211bf410bfc29bf/Qwen3.6-27B-Q4_K_M.gguf \
        -md /home/livens/.cache/huggingface/hub/models--spiritbuun--Qwen3.5-27B-DFlash-GGUF/snapshots/3fa59f082214838d36d01d5d1276758efb1f3b3c/dflash-draft-q4_k_m.gguf \
     --spec-type dflash \
    --cache-type-k q4_0 --cache-type-v q4_0 \
    -ngl 999 -c 131072 --threads 2 \
    -fa on --cache-ram 0 \
    --jinja  \
    --temp 1.0 --top-p 0.95 --top-k 20 \
    --host 0.0.0.0 --port 11434

### Relevant log output

srv          init: init: chat template, thinking = 1
main: model loaded
main: server is listening on http://0.0.0.0:11434
main: starting the main loop...
srv  update_slots: all slots are idle
srv  log_server_r: done request: GET / 192.168.6.147 200
srv  log_server_r: done request: GET /bundle.css 192.168.6.147 200
srv  log_server_r: done request: GET /bundle.js 192.168.6.147 200
srv  log_server_r: done request: HEAD /cors-proxy 192.168.6.147 404
srv  params_from_: Chat format: peg-native
slot get_availabl: id  3 | task -1 | selected slot by LRU, t_last = -1
slot launch_slot_: id  3 | task -1 | sampler chain: logits -> ?penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> min-p -> ?xt> ?temp-ext -> dist
slot launch_slot_: id  3 | task 0 | processing task, is_child = 0
slot update_slots: id  3 | task 0 | new prompt, n_ctx_slot = 131072, n_keep = 0, task.n_tokens = 15
slot update_slots: id  3 | task 0 | n_tokens = 0, memory_seq_rm [0, end)
srv  log_server_r: done request: POST /v1/chat/completions 192.168.6.147 200
slot update_slots: id  3 | task 0 | prompt processing progress, n_tokens = 11, batch.n_tokens = 11, progress = 0.733333
slot update_slots: id  3 | task 0 | n_tokens = 11, memory_seq_rm [11, end)
reasoning-budget: activated, budget=2147483647 tokens
slot init_sampler: id  3 | task 0 | init sampler, took 0.01 ms, tokens: text = 15, total = 15
slot update_slots: id  3 | task 0 | prompt processing done, n_tokens = 15, batch.n_tokens = 4
sched_reserve: reserving ...
sched_reserve:      CUDA0 compute buffer size =    61.88 MiB
sched_reserve:  CUDA_Host compute buffer size =     4.08 MiB
sched_reserve: graph nodes  = 209
sched_reserve: graph splits = 4
sched_reserve: reserve took 10.78 ms, sched copies = 1
dflash draft breakdown (ctx=5): concat=0.4ms cross=0.8ms decode=88.4ms argmax=0.0ms total=89.6ms
allocate_tree_buffers: allocated tree verify buffers: 48 layers × 20 tokens = 1440.0 MB
allocate_tape_gpu: allocated GPU tape buffer: 30.4 MB (48 layers, 20 max tokens)
dflash draft breakdown (ctx=10): concat=0.8ms cross=0.1ms decode=66.9ms argmax=0.0ms total=67.8ms
init: the tokens of sequence 3 in the input batch have inconsistent sequence positions:
 - the last position stored in the memory module of the context (i.e. the KV cache) for sequence 3 is X = 30
 - the tokens for sequence 3 in the input batch have a starting position of Y = 20
 for M-RoPE, it is required that the position satisfies: X <= Y
decode: failed to initialize batch
llama_decode: failed to decode, ret = -1
srv  update_slots: Invalid input batch. i = 0, n_batch = 256, ret = -1
srv    send_error: task id = 0, error: Invalid input batch.
slot      release: id  3 | task 0 | stop processing: n_tokens = 36, truncated = 0
slot prompt_clear: id  3 | task -1 | clearing prompt with 36 tokens
srv          stop: cancel task, id_task = 0
srv  update_slots: all slots are idle

**Comment by shirubei** [comment]
similar error. decrease context size may work.
